src/ssb_project_cli/ssb_project/create/github.py

```diff
@@ -192,7 +200,98 @@ def set_branch_protection_rules(
         repo_name: name of repository
         github_org_name: Name of GitHub organization
     """
-    repo = Github(github_token).get_repo(f"{github_org_name}/{repo_name}")
+    repo = get_environment_specific_github_object(github_token).get_repo(
+        f"{github_org_name}/{repo_name}"
+    )
     repo.get_branch("main").edit_protection(
         required_approving_review_count=1, dismiss_stale_reviews=True
     )
+
+
+def get_environment_specific_github_object(github_token: str) -> Github:
+    """Creates and returns a `Github` object with appropriate settings based on the environment.
+
+    Args:
+        github_token: A personal access token for authenticating with the GitHub API.
+
+    Returns:
+        A `Github` object that can be used to interact with the GitHub API.
+
+    This function creates a `Github` object that is specific to the current environment.
+    If the function is running in the onprem environment, SSL verification uses /etc/ssl/certs/ca-certificates.crt.
+    Otherwise, SSL verification is enabled.
+    """
+    if running_onprem(JUPYTER_IMAGE_SPEC):
+        # CA bundle to use, supplying this fixes the onprem error "CERTIFICATE_VERIFY_FAILED"
+        # verify can be boolean or string, we have to type ignore because mypy expects it to be a bool
+        return Github(github_token, verify="/etc/ssl/certs/ca-certificates.crt")  # type: ignore
+    else:
+        return Github(github_token)
+
+
+def get_org_members(github_token: str) -> list[str]:
+    """Returns a list of login names for all members of a GitHub organization.
+
+    Args:
+        github_token: GitHub API token.
+
+    Returns:
+        list: A list of strings, where each string is the login name of a member of the organization.
+    """
+    # Set up the API endpoint URL and initial query parameters
+    url = f"https://api.github.com/orgs/{GITHUB_ORG_NAME}/members"
+    params = {"per_page": 100, "page": 1}
+    headers = {"Authorization": f"Bearer {github_token}"}
+
+    # Store usernames
+    github_usernames = []
+
+    while True:
+        response = requests.get(
+            url,
+            headers=headers,
+            params=params,
+            timeout=20,
+            verify="/etc/ssl/certs/ca-certificates.crt",
+        )
+
+        if response.status_code == 200:
+            members = response.json()
+            if len(members) == 0:
+                break
+
+            for member in members:
+                github_usernames.append(member["login"])
+            params["page"] += 1
+        else:
+            print("Error: could not retrieve member list")
```

Comment:
Could we record the response for debugging purposes?

Reply:
Thats a good idea.

Added logs for response with our create_error_log function. In addition we now store debug logs for the latest run in .error_logs/ssb-project-debug.log. requests and pygithub has very good logs, this will be useful the next time they change up network rules onprem.